pymc/tests/test_model.py

```diff
@@ -702,6 +703,78 @@ def test_nested_model_coords():
     assert set(m2.RV_dims) < set(m1.RV_dims)
 
 
+def test_shape_error_from_fixed_dims():
+    """
+    with #5751 dimensions set by add_coord() are default TensorConstant type
+    and so have fixed dimensions. Trying to resize them will raise a ShapeError,
+    which this test checks for.
+    """
+    with pm.Model() as pmodel:
+        a = pm.Normal("a", mu=[1, 2, 3], dims="fixed")
+
+        m = pm.MutableData("m", [[1, 2, 3]], dims=("one", "fixed"))
+
+        # This is fine because the "fixed" dim is not resized
+        pm.set_data({"m": [[1, 2, 3], [3, 4, 5]]})
+
+    with pytest.raises(ShapeError, match="was initialized from 'a'"):
+        # Can't work because the "fixed" dimension is linked to a constant shape:
+        # Note that the new data tries to change both dimensions
+        with pmodel:
+            pm.set_data({"m": [[1, 2], [3, 4]]})
+
+
+def test_shape_warning_from_non_fixed_dims():
+    """
+    test that resizing dims gives a ShapeWarning.
+    """
+    with pm.Model() as pmodel:
+        pmodel.add_coord("shared", [1, 2, 3], fixed=False)
+        m = pm.MutableData("m", [1, 2, 3], dims=("shared"))
+    with pytest.warns(ShapeWarning, match="was initialized from coords"):
+        with pmodel:
+            # correctly passes coords during resize and so warns user with ShapeWarning
+            pm.set_data({"m": [1, 2, 3, 4]}, coords={"shared": [1, 2, 3, 4]})
+
+
+def test_value_error_from_non_fixed_dims_no_coords():
+    """
+    test that resizing dims without passing coords raises a ValueError.
+    """
+    with pm.Model() as pmodel:
+        pmodel.add_coord("shared", [1, 2, 3], fixed=False)
+        m = pm.MutableData("m", [1, 2, 3], dims=("shared"))
+    with pytest.raises(ValueError, match="'m' variable already had 3"):
+        with pmodel:
```

Comment:
No need to close & open the modelcontext
```suggestion
        with pytest.raises(ValueError, match="'m' variable already had 3"):
```